src/main/java/org/truffleruby/core/exception/CoreExceptions.java

```diff
@@ -930,28 +895,6 @@ public DynamicObject systemCallError(String message, int errno, Node currentNode
         return ExceptionOperations.createSystemCallError(context, exceptionClass, errorMessage, currentNode, errno);
     }
 
-    // IO::EAGAINWaitReadable, IO::EAGAINWaitWritable
-
-    @TruffleBoundary
-    public DynamicObject eAGAINWaitReadable(Node currentNode) {
-        DynamicObject exceptionClass = context.getCoreLibrary().getEagainWaitReadable();
-        return ExceptionOperations.createSystemCallError(
-                context,
-                exceptionClass,
-                coreStrings().RESOURCE_TEMP_UNAVAIL_READ.createInstance(),
```

Comment:
Probably `RESOURCE_TEMP_UNAVAIL_READ` becomes dead due to this, right?
Then it can be removed as well.

Reply:
I'll leave it as-is for now to be able to merge this. But I will do another round in a few days (also adding the guards as Chris mentions below), then this will be included.